sympy/physics/units/dimensions.py

```diff
@@ -439,7 +439,7 @@ def get_dimensional_dependencies(self, name, mark_dimensionless=False):
         dimdep = self._get_dimensional_dependencies_for_name(name)
         if mark_dimensionless and dimdep == {}:
             return {Dimension(1): 1}
-        return {k: v for k, v in dimdep.items()}
+        return dict(dimdep.items())
```

Comment:
I wonder what the purpose of this lne is. If `dimdep` is a dict then is there really any reason we can't just return it? At least if we want a copy then that's just `dict(dimdep)` or `dimdep.copy()`.

Reply:
I suspect that this kind of things comes from continuous edits to the code that end up with something sloppy that didn't really need to be there. This is probably the sort of thing where a linter rule (if enabled when the code was last edited) would have prompted the author to simplify this code.

Reply:
Don't know without a hint/strong typing I am afraid. Could be a modified class or so. 
Hence, I chose the safe solution.

Reply:
> This is probably the sort of thing where a linter rule (if enabled when the code was last edited) would have prompted the author to simplify this code.

Which is why I share your point of view that exploring the linter - rather than silencing it by pinning to old versions - could gradually improve legacy issues.